core/src/backend/navigator.rs

```diff
@@ -111,6 +121,14 @@ impl Request {
     pub fn set_body(&mut self, body: (Vec<u8>, String)) {
         self.body = Some(body);
     }
+
+    pub fn headers(&self) -> &[(String, String)] {
+        &self.headers
+    }
+
+    pub fn set_headers(&mut self, headers: Vec<(String, String)>) {
```

Comment:
nit: Maybe we should set a bit simpler interface? How about `add_header`?

Reply:
I've changed the code to perform header name deduplication in `loader.rs`. The simplest interface is now for `set_headers` to take in an `IndexMap`.